source/extensions/outlier_detection_monitors/consecutive_errors/consecutive_errors_monitor.cc

```diff
@@ -0,0 +1,50 @@
+#include "source/extensions/outlier_detection_monitors/consecutive_errors/consecutive_errors_monitor.h"
+
+#include "envoy/extensions/outlier_detection_monitors/consecutive_errors/v3/consecutive_errors.pb.h"
+#include "envoy/extensions/outlier_detection_monitors/consecutive_errors/v3/consecutive_errors.pb.validate.h"
+#include "envoy/registry/registry.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Outlier {
+
+bool ConsecutiveErrorsMonitor::onError() {
+  if (counter_ < max_) {
+    counter_++;
+  }
+
+  return (counter_ == max_);
+}
+
+void ConsecutiveErrorsMonitor::onSuccess() {
+  // start counting from zero again.
+  counter_ = 0;
+}
+
+void ConsecutiveErrorsMonitor::onReset() { counter_ = 0; }
+
+class ConsecutiveErrorsMonitorFactory
```

Comment:
Most of Envoy extensions follow a specific code layout, this is not enforced but helps to easier navigate the code, group tests etc. The factories that parse the extension proto config and create extension instances are usually placed into `config.h+cc` files. Extension related code goes into its dedicated header+impl files. For example:
[extension config](https://github.com/envoyproxy/envoy/blob/main/source/extensions/resource_monitors/downstream_connections/config.h)
[extension](https://github.com/envoyproxy/envoy/blob/main/source/extensions/resource_monitors/downstream_connections/downstream_connections_monitor.h)

Reply:
I dont have string opinion here since some extensions deviate from this layout and this file does not have that much code.

Reply:
OK. Thanks. I will leave it for now as it is.